Tools/AP_Periph/serial_options.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  serial options support, for serial over DroneCAN
+ */
+
+#include "AP_Periph.h"
+
+#ifdef HAL_PERIPH_ENABLE_SERIAL_OPTIONS
+
+#include "serial_options.h"
+#include <AP_SerialManager/AP_SerialManager_config.h>
+
+extern const AP_HAL::HAL &hal;
+
+const AP_Param::GroupInfo SerialOptions::var_info[] {
+
+#if HAL_HAVE_SERIAL0
+    // @Group: 0_
+    // @Path: serial_options_dev.cpp
+    AP_SUBGROUPINFO(devs[0], "0_",  1, SerialOptions, SerialOptionsDev),
+#endif
+
+#if HAL_HAVE_SERIAL1
+    // @Group: 1_
+    // @Path: serial_options_dev.cpp
+    AP_SUBGROUPINFO(devs[1], "1_",  2, SerialOptions, SerialOptionsDev),
+#endif
+
+#if HAL_HAVE_SERIAL2
+    // @Group: SERIAL2_
```

Comment:
Serial2_ should be 2_ shouldn't this be breaking the param checker for name length?

Reply:
thanks! fixed